app/models/concerns/orchestration.rb

```diff
@@ -31,8 +31,16 @@ def register_rebuild(method, pretty_name)
 
   protected
 
-  def on_save
+  def around_save_orchestration
     process :queue
+
+    begin
+      yield
+    rescue ActiveRecord::ActiveRecordError => e
+      logger.warn "Rolling back due to exception during save: #{e}"
```

Comment:
Prefer Foreman::Logging.exception when logging exceptions to get the backtrace etc.  You can pass the warn level in as a parameter.